deeposlandia/webapp.py

```diff
@@ -0,0 +1,153 @@
+
+
+@app.route("/predictor_demo/<string:model>/<string:dataset>")
+def predictor_demo(model, dataset):
+    check_model(model)
+    check_dataset(dataset)
+    if dataset == "shapes":
+        sample_filename = os.path.join("sample_image", "shape_example.png")
+        return render_template('shape_demo.html', model=model,
+                               sample_filename=sample_filename)
+    else:
+        sample_filename = os.path.join("sample_image", "example.jpg")
+        return render_template('mapillary_demo.html', model=model,
+                               sample_filename=sample_filename)
+
+
+@app.route("/predictor")
+def predictor():
+    filename = os.path.join("sample_image", "example.jpg")
+    return render_template("predictor.html", example_image=filename)
+
+
+@app.route("/demo_prediction")
+def demo_prediction():
+    filename = request.args.get('img')
+    filename = os.path.join("deeposlandia", filename[1:])
+    dataset = request.args.get('dataset')
+    model = request.args.get('model')
+    utils.logger.info("file: {}, dataset: {}, model: {}".format(filename,
+                                                                dataset,
+                                                                model))
+    if model == "feature_detection":
+        predictions = predict([filename], dataset, model)
+        return jsonify(predictions)
+    elif model == "semantic_segmentation":
+        predictions = predict([filename], dataset, model,
+                              output_dir=PREDICT_FOLDER)
+        return jsonify(predictions)
+    else:
+        utils.logger.error(("Unknown model. Please choose "
+                            "'feature_detection' or 'semantic_segmentation'."))
+        return ""
+
+
+@app.route("/prediction")
+def prediction():
+    filename = request.args.get('img').split("/")[-1]
+    filename = os.path.join(app.config['UPLOAD_FOLDER'], filename)
+    dataset = request.args.get('dataset')
+    model = request.args.get('model')
+    utils.logger.info("file: {}, dataset: {}, model: {}".format(dataset, model,
+                                                                filename))
+    predictions = predict([filename], "mapillary", "semantic_segmentation",
+                          aggregate=False, output_dir=PREDICT_FOLDER)
+    return jsonify(predictions)
+
+
+@app.route('/uploads/<filename>')
+def send_image(filename):
+    return send_from_directory(app.config['UPLOAD_FOLDER'], filename)
+
+
+@app.route("/predictor", methods=['GET', 'POST'])
+def upload_image():
+    if request.method == 'POST':
+        # check if the post request has the file part
+        if 'file' not in request.files:
+            print('No file part')
+            return redirect(request.url)
+        file = request.files['file']
+        # if user does not select file, browser also
+        # submit a empty part without filename
+        if file.filename == '':
+            print('No selected file')
```

Comment:
remaining print

Reply:
Actually this is a stackoverflow-oriented development issue, there... Maybe we can keep the log as the function behavior still have to be clarified. I replace `print` statements with `utils.logger.info`.